### ISSUE TITLE
Lock screen bypassed by plugging in a monitor

### ISSUE THREAD

**Issue by folz** [linuxmint/cinnamon-screensaver#246]
```
 * Cinnamon-Screensaver version: 3.4.2
 * Distribution - Linux Mint 18.2 Sonya
 * Graphics hardware *and* driver used: nvidia-375 proprietary drivers on GeForce 840M
 * 32 or 64 bit: 64bit x86_64
 ```

**Issue**

Plugging in a monitor causes the desktop display to resize, but screensaver does not resize. This allows me to interact with my desktop without entering my password.

1-minute video demonstrating the issue: https://www.youtube.com/watch?v=EWs_SQJ1-Ak

(I call this a LightDM bypass in the video because I didn't know about cinnamon-screensaver and thought lightdm provided the lock screen.)

**Steps to reproduce**

Plug in my monitor.

**Expected behaviour**

The entire screen, including the new monitor, remains locked.

**Other information**

I started getting this issue after I upgraded to 18.2 and switched from MDM to LightDM, but this may be a red herring as I don't remember if cinnamon-screensaver was also upgraded during this process.

My laptop display is a 3200x1800 QHD (HiDPI) display when it's standalone but I downscale it to 1600x900 (non-HiDPI) when it has an attached monitor, due to scaling issues when connecting HiDPI and regular monitors. It's possible that the lock screen doesn't think to resize, or resizes incorrectly, when the laptop screen automatically downscales as the new display is attached.

**Comment by folz** [comment]
Looking through the issues, this is probably related to #239 and #244, as they all have a common thread of the virtual display resizing and cinnamon-screensaver not matching the resize.

**Comment by kykrueger** [comment]
I was just wondering if this would happen like the VM problem.

**Comment by mtwebster** [comment]
Hi,

If possible, I'd like you to test some packages for me.  You need to be up-to-date to level 3 updates (which include all cinnamon-related updates).  These are for Mint 18.2 only, 64 bit.

Download the following (but do not try to open with the installer):
https://www.dropbox.com/s/68awx5plb5aglr9/cinnamon-screensaver_3.4.2_all.deb?dl=0
https://www.dropbox.com/s/b4s6og1aen2ydul/cinnamon-screensaver-pam-helper_3.4.2_amd64.deb?dl=0
https://www.dropbox.com/s/xqizghg2u7lfzrz/libcscreensaver0_3.4.2_amd64.deb?dl=0
https://www.dropbox.com/s/wdxzmatca19wrfs/libcscreensaver-dbg_3.4.2_amd64.deb?dl=0

optional (only if you already have these installed for extra screensavers):
https://www.dropbox.com/s/kjob1moeat2scza/cinnamon-screensaver-x-plugin_3.4.2_all.deb?dl=0
https://www.dropbox.com/s/321qdp7mg3newbz/cinnamon-screensaver-webkit-plugin_3.4.2_all.deb?dl=0

As I said, download them all, then open a terminal in the folder you saved them to, and:

`sudo dpkg -i *screen*.deb`

Then kill and restart cinnamon-screensaver, or log out and back in.  I'm hoping I've got this 'family' of issues worked out, in virtual machines and real hardware, and would appreciate a bit of feedback before I try to get a point release.

PS: The update manager will want overwrite these with the current repository versions, so you'll either have to uncheck them if you update, or right-click->ignore them for now.

PPS: Sorry about the crappy dropbox links, they recently changed how they do file sharing, I'm looking for an alternative.

Thanks!

**Comment by kykrueger** [comment]
Hi Michael, I'll try and test these out today. Thanks for looking into this issue.-- Kyle                                                                                                                                                                                                                                                                                                                 Von: notifications@github.comGesendet: 18. September 2017 6:50 vorm.An: cinnamon-screensaver@noreply.github.comAntworten: reply@reply.github.comCc: kyle.s.krueger@gmail.com; comment@noreply.github.comBetreff: Re: [linuxmint/cinnamon-screensaver] Lock screen bypassed by plugging in a monitor (#246)  Hi,
If possible, I'd like you to test some packages for me.  You need to be up-to-date to level 3 updates (which include all cinnamon-related updates).  These are for Mint 18.2 only, 64 bit.
Download the following (but do not try to open with the installer):
https://www.dropbox.com/s/68awx5plb5aglr9/cinnamon-screensaver_3.4.2_all.deb?dl=0
https://www.dropbox.com/s/b4s6og1aen2ydul/cinnamon-screensaver-pam-helper_3.4.2_amd64.deb?dl=0
https://www.dropbox.com/s/xqizghg2u7lfzrz/libcscreensaver0_3.4.2_amd64.deb?dl=0
https://www.dropbox.com/s/wdxzmatca19wrfs/libcscreensaver-dbg_3.4.2_amd64.deb?dl=0
optional (only if you already have these installed for extra screensavers):
https://www.dropbox.com/s/kjob1moeat2scza/cinnamon-screensaver-x-plugin_3.4.2_all.deb?dl=0
https://www.dropbox.com/s/321qdp7mg3newbz/cinnamon-screensaver-webkit-plugin_3.4.2_all.deb?dl=0
As I said, download them all, then open a terminal in the folder you saved them to, and:
sudo dpkg -i *screen*.deb
Then kill and restart cinnamon-screensaver, or log out and back in.  I'm hoping I've got this 'family' of issues worked out, in virtual machines and real hardware, and would appreciate a bit of feedback before I try to get a point release.
PS: The update manager will want overwrite these with the current repository versions, so you'll either have to uncheck them if you update, or right-click->ignore them for now.
PPS: Sorry about the crappy dropbox links, they recently changed how they do file sharing, I'm looking for an alternative.
Thanks!

—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by mtwebster** [comment]
Any updates from people that have tried the packages above?  Can I assume no response means it's working so well you forgot all about mentioning anything? ;)

**Comment by Piiit** [comment]
@mtwebster Hi, sorry haven't had the time to test. Hopefully, I will soon...

**Comment by anopse** [comment]
@mtwebster I had the same issue and after a few tests I can't reproduce the bug anymore using your deb packages. Because the behavior always has been inconsistent I wouldn't trust only 1 feedback but for now it's working fine. :+1:

**Comment by vascotenner** [comment]
@mtwebster: with your new packages I cannot reproduce this issue.

**Comment by kykrueger** [comment]
@mtwebster It has been a long time since I have seen this issue, but just saw a glimpse of it again today on a native linux-mint 19 (tara) installation. Is it possible that it has regressed?

To be clear, the issue I saw that was when docking a locked laptop onto a docking station, I saw the open applications for a very very short time. I wasn't able to check if it was possible to kill the lock-screen by pasting something into one of the terminals.
